Formula/node@14.rb

```diff
@@ -26,28 +26,30 @@ class NodeAT14 < Formula
   # disable! date: "2023-04-30", because: :unsupported
 
   depends_on "pkg-config" => :build
+  depends_on "python@3.10" => :build
   depends_on "brotli"
   depends_on "c-ares"
   depends_on "icu4c"
   depends_on "libnghttp2"
   depends_on "libuv"
   depends_on "openssl@1.1"
 
-  uses_from_macos "python"
   uses_from_macos "zlib"
 
   on_macos do
-    depends_on "python@3.10" => [:build, :test]
     depends_on "macos-term-size"
   end
 
-  def python3
-    Formula["python@3.10"]
+  on_linux do
+    # npm with node-gyp>=8.0.0 is needed for Python 3.11 support
+    # Issue ref: https://github.com/nodejs/node-gyp/issues/2219
+    # Commit ref: https://github.com/nodejs/node-gyp/commit/9e1397c52e429eb96a9013622cffffda56c78632
+    depends_on "python@3.10"
   end
```

Comment:
Node 14 ships with npm 6. Lastest version has package.json with `"node-gyp": "^5.1.1"`. I doubt this will get a major update or have necessary backports for Python 3.11 support before EOL.

On side note, may need to figure out how to add an env script for Linux.